### PR TITLE
build: stop detecting concepts support

### PR DESCRIPTION
**Pull request by tchaikov**
* remove all stuff use for detecting C++20 concepts

since we only support the latest two major releases of compilers. at the moment of writing, Clang just released v15, and the latest major release of GCC is v12. both of them have a decent support of C++ concepts. and if the tree is compiled with `-std=c++20`, `-std=c++2a` or `-std=c++23`, `-fconcepts` or `-fconcepts-ts` is not necessary for compiling C++ code using C++20 concepts.

also, because neither libc++ nor libstdc++ defines `__cpp_lib_concepts` if the source is compiled with `-std=c++<standard>` where "standard" is lower than 20, `include/seastar/util/concepts.hh` would define `SEASTAR_CONCEPT` to nothing. also, `-std=c++17` makes compilers define `__cpp_concepts` as `201507`. this also practically disables `SEASTAR_CONCEPT`. please note, `__cpp_concepts` only detects the compiler's capability for supporting C++20 concepts, but we still need the support from standard library to program using C++20 concepts.

also, `code_tests/Concepts_test.cc` compares `__cpp_concepts` with `201507`. this always fails when compiling with C++20, where `__cpp_concepts` is `201907L` or `202002L`. see
https://isocpp.org/std/standing-documents/sd-6-sg10-feature-test-recommendations#__cpp_concepts .

so, in short, we don't use concepts when the tree is compiled with C++17. and when compiling with C++20, supported C++ compilers do not need an additional compiling option for enabling concepts support.

before this change, we pass `-fconcepts` to compiler if it compiles code using C++20 concepts.

after this change, we stop checking the C++20 concepts support and do not pass `-fconcepts` to compiler even it is accepted by the compiler when compiling the test code using C++20 concepts.

Signed-off-by: Kefu Chai <kefu.chai@scylladb.com>

### CODE REVIEW

**Comment by avikivity** [comment]
We still support C++17. And C++17 does not support concepts (even if compilers do as an extension, which they shouldn't).

**Comment by avikivity** [comment]
Or is this only in C++20 mode?

**Comment by tchaikov** [comment]
exactly, we only enable concepts when the tree is compiled with C++20. so C++17 support is intact.

**Comment by tchaikov** [comment]
@avikivity hi Avi, did i address your concern?
